controllers/dnsupdate.go

```diff
@@ -43,9 +43,19 @@ func (r *GslbReconciler) gslbDNSEndpoint(gslb *k8gbv1beta1.Gslb) (*externaldns.D
 		return nil, err
 	}
 
-	localTargets, err := r.DNSProvider.GslbIngressExposedIPs(gslb)
-	if err != nil {
-		return nil, err
+	localTargets := []string{}
+	if gslb.Spec.Ingress != nil {
+		localTargets, err = r.DNSProvider.GslbIngressExposedIPs(gslb)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if gslb.Spec.LoadBalancer != nil {
+		localTargets, err = r.DNSProvider.GslbServiceExposedIPs(gslb)
```

Comment:
this way, if both `gslb.spec.loadBalancer` and `gslb.spec.ingress` are specified in the CR, then the `localTargets` will be obtained using the LB section. I am not sure how it should be designed, to be honest. Does it actually make sense that gslb CR is backed by both at once? If not we could implement the [`x-kubernetes-validations`](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2876-crd-validation-expression-language/README.md) to catch this during resource admission, otherwise I'd go with simple `if - else if` here for now.

Reply:
yeah, I think adding a validation for it is better so that both can't be specified. It is a cleaner and better approach then making some blunders while missing some case in if - else approach. It is better to be clear at the start of the implementation and state that both can't be used at once. WDYT?

Reply:
makes sense to me. Would you mind exploring those [x-kubernetes-validations](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2876-crd-validation-expression-language/README.md)? There should be an annotation for kubebuilder for that ([link](https://github.com/kubernetes-sigs/kubebuilder/issues/2798#issuecomment-1310275396))

something like 

```
// +kubebuilder:validation:XValidation:rule=( has(self.loadBalancer) && !has(self. ingress) ) || ( !has(self.loadBalancer) && has(self. ingress) )
```